### PR TITLE
solve raises error on duplicate symbols

### PR DESCRIPTION
**Pull request by smichr**
#### Brief description of what is fixed or changed

Rather than deal in some unique way with a solve request that contains duplicate symbols, the presence of duplicates now raises an error. This seems like a good idea because there is no reason to include duplicates and it *might* indicate a typo of the person entering the request. Disallowing duplicates matches the behvior of `nonlinsolve` and `linsolve`

#### Other comments

There are lots of stray extra lines fixed in the first commit. Please review commits starting at 2 instead of looking at "Files changed" if that much noise is bothersome.

#### Release Notes

* solvers
  * `solve` now raises an error if duplicate symbols are given in the unknowns to be solved for such as `solve(equations, [x, y, x])`.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This doesn't catch the case of duplicated symbols when the input is not a list:
```python
In [4]: solve(x, x, y, x)
Out[4]: [{x: 0}, {x: 0}]

In [5]: solve([x], x, y, x)
---------------------------------------------------------------------------
GeneratorsError
```

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -1016,8 +1018,7 @@ def _sympified_list(w):
 
     # we can solve for non-symbol entities by replacing them with Dummy symbols
     f, symbols, swap_sym = recast_to_symbols(f, symbols)
-
-    # this is needed in the next two events
+    # udpate symset
```
Comment (smichr) [inline sympy/solvers/solvers.py:1021]:
```suggestion
    # this set of symbols (perhaps recast) is needed below
```

**Comment by smichr** [comment]
Thanks for the catch, @oscarbenjamin. This is ready to go.

**Comment by oscarbenjamin** [comment]
Looks good!

**Pull request merged**